### ISSUE TITLE
(OpenWrt) v2raya (1.5.9.1698.1) incompatibility with v2ray-core 5.1.0

### ISSUE THREAD

**Issue by lss4** [v2rayA/v2rayA#676]
TL;DR workaround: Use the `luci-app-v2raya` package to specify your own copy of `v2ray` binary. With the right version you can work this issue around, and there's no need to actually install a copy of `v2ray-core` using `opkg` anymore.

------

### Version, approach of installation, operating system

1. What version of v2rayA are you using

1.5.9.1698.1

2. What approach of installation do you install V2rayA?

Image created using OpenWrt ImageBuilder with v2rayA built-in (using the official v2raya opkg repo).

3. What is your operating system?

OpenWrt Snapshot (r20862) built on Oct 5, 2022.

### Description

v2rayA from official OpenWrt repo at the time of building the image, cannot be used with `v2ray-core` 5.1.0 due to #652, per [this comment](https://github.com/v2rayA/v2rayA/issues/652#issuecomment-1271102703).

However, I can still start v2ray-core manually for temporary use, using the config file previously generated by v2rayA (`/etc/v2raya/config.json`). Due to system space constraints I cannot move the `v2ray` binary away to replace it with a wrapper like the comment I mentioned above.

Should also mention that due to v2ray exiting unexpectedly thanks to the incompatible command line, even saving proxy settings will fail with the same "unexpected exiting" error.

### Steps to reproduce

- Open v2rayA web page.
- Try starting the proxy.
- Proxy will not start with an "unexpected exiting" error.

### Logs：

```
2022/10/08 10:43:00.597  [I] [io.go:428]  flag provided but not defined: -config
```

### v2rayA setting and related screenshot

N/A

**Comment by Ray6899** [comment]
有临时的解决方案吗

**Comment by lss4** [comment]
It's been a month since I wrote this issue. Looks like v2rayA's OpenWrt repo had [another update](https://github.com/v2rayA/v2raya-openwrt/commit/dfbc884495f4440b3d6a2978ff8fe69ca39a346f) recently (4 days ago as I'm writing), and the `v2ray-core` package in the repo is now at 4.45.2 which works with the latest v2rayA release.

I'll build a new image when I have time to confirm if the new image will indeed have this working version of `v2ray-core`.

NOTE: If you installed `v2ray-core` and `v2raya` by hand using `opkg` (assuming your device has enough storage), you can simply try uninstalling your current `v2ray-core`, then reinstall it, to see if you're getting the 4.45.2 version instead of 5.1.0.

**Comment by Ray6899** [comment]
Thank you, I will try.

**Comment by lss4** [comment]
Just tried building another image and it's still pulling `v2ray-core` 5.1.0, which actually came from the [official OpenWrt repo](https://downloads.openwrt.org/snapshots/packages/arm_cortex-a15_neon-vfpv4/packages/v2ray-core_5.1.0-2_arm_cortex-a15_neon-vfpv4.ipk).

So there was nothing wrong with the v2rayA repo after all. The issue is actually caused by the 5.1.0 version of `v2ray-core` in the official repo which is taking precedence over the 4.45.2 version in v2rayA's repo when building the image.

**Comment by lss4** [comment]
UPDATE: So far I haven't been able to successfully make ImageBuilder use `v2ray-core` 4.45.2 instead of 5.1.0.

Apparently there's another package called `luci-app-v2raya` in the repo. You can use to specify the path to your own `v2ray` binary to use instead of the one you installed into the image (or via opkg). With the right version you can work this issue around.

NOTE: With this new feature from `luci-app-v2raya`, actually installing a copy of `v2ray-core` via `opkg` is no longer necessary.
